spring-pulsar-docs/src/main/antora/modules/ROOT/pages/reference/pulsar/transactions.adoc

```diff
@@ -0,0 +1,222 @@
+The following example shows how to set the timeout on the auto-configured template:
+
+[source, java]
+----
+@Bean
+PulsarTemplateCustomizer templateCustomizer() {
+    return (template) -> template.transactions().setTimeout(Duration.ofSeconds(45));
+}
+----
+
+== Transactional Receiving with `@PulsarListener`
+When listener transactions are enabled, the `@PulsarListener` annotated listener method is invoked in the scope of a synchronized transaction.
+
+The `DefaultPulsarMessageListenerContainer` uses a Spring `TransactionTemplate` configured with a `PulsarTransactionManager` to initiate the transaction prior to method invocation.
+
+The acknowledgment of each received message is enlisted in the scoped transaction.
+
+=== Consume-Process-Produce Scenario
+A common transactional pattern is where a consumer reads messages from a Pulsar topic, transforms the messages, and finally a producer writes the resulting messages to another Pulsar topic.
+The framework supports this use case when transactions are enabled and your listener method uses a transactional `PulsarTemplate` to produce the transformed message.
+
+Given the following listener method:
+[source, java]
+----
+@PulsarListener(topics = "my-input-topic") // <1>
+void listen(String msg) { // <2>
+    var transformedMsg = msg.toUpperCase(); // <3>
+    this.transactionalTemplate.send("my-output-topic", transformedMsg); // <4>
+} // <5> <6>
+----
+The following interactions occur when listener transactions are enabled:
+
+<1> Listener container initiates new transaction and invokes listener method in scope of transaction
+<2> Listener method receives message
+<3> Listener method transforms message
+<4> Listener method sends transformed message with transactional template which enlists send operation in active transaction
+<5> Listener container auto-acks message and enlists ack operation in active transaction
+<6> Listener container (via `TransactionTemplate`) commits transaction
+
+If you are not using `@PulsarListener` and instead using listener containers directly, the same transaction support is provided as described above.
+Remember, the `@PulsarListener` is just a convenience to register a Java method as the listener container message listener.
+
+=== Transactions with Record Listeners
+The above example uses a record listener.
+When using a record listener, a new transaction is created on every listener method invocation which equates to a transaction per message.
+
+NOTE: Because the transaction boundary is per message and each message acknowledgement is enlisted in each transaction, batch ack mode  can not be used with transactional record listeners.
+
+=== Transactions with Batch Listeners
+When using a batch listener, a new transaction is created on every listener method invocation which equates to a transaction per batch of messages.
+
+NOTE: Due to complexity, transactional batch listeners do not currently support custom error handlers.
```

Comment:
I don't think we need to say "Due to complexity".   Rather just say, "Transactional batch listeners do not currently support custom error handlers."